packages/features/form-builder/FormBuilder.tsx

```diff
@@ -119,29 +125,37 @@ export const FormBuilder = function FormBuilder({
         <p className="text-subtle mt-0.5 max-w-[280px] break-words text-sm sm:max-w-[500px]">{description}</p>
         <ul ref={parent} className="border-subtle divide-subtle mt-4 divide-y rounded-md border">
           {fields.map((field, index) => {
-            const options = field.options
-              ? field.options
-              : field.getOptionsAt
-              ? dataStore.options[field.getOptionsAt as keyof typeof dataStore]
-              : [];
-
-            // Note: We recently started calling getAndUpdateNormalizedValues in the FormBuilder. It was supposed to be called only on booking pages earlier.
-            // Due to this we have to meet some strict requirements like of labelAsSafeHtml.
+            let options = field.options ?? null;
+            const sources = [...(field.sources || [])];
+            const isRequired = isRequiredField(field);
+            if (!options && field.getOptionsAt) {
+              options = dataStore.options[field.getOptionsAt as keyof typeof dataStore] ?? [];
+              // TODO: The dataStore should itself provide the label directly
+              // This is important because FormBuilder isn't aware of what a location is. It is a generic Form Builder
+              const sourceLabel = field.getOptionsAt === "locations" ? "Location" : field.getOptionsAt;
+              options.forEach((option) => {
+                sources.push({
+                  id: option.value,
+                  label: sourceLabel,
+                  type: "system",
+                });
+              });
+            }
+
             if (fieldsThatSupportLabelAsSafeHtml.includes(field.type)) {
               field = { ...field, labelAsSafeHtml: markdownToSafeHTML(field.label ?? "") };
             }
-
-            const { hidden } = getAndUpdateNormalizedValues({ ...field, options }, t);
-            if (field.hideWhenJustOneOption && (hidden || !options?.length)) {
+            const numOptions = options?.length ?? 0;
+            const firstOptionInput =
+              field.optionsInputs?.[options?.[0]?.value as keyof typeof field.optionsInputs];
+            const doesFirstOptionHaveInput = !!firstOptionInput;
+            // If there is only one option and it doesn't have an input required, we don't show the Field for it.
+            // Because booker doesn't see this in UI, there is no point showing it in FormBuilder to configure it.
+            if (field.hideWhenJustOneOption && numOptions <= 1 && !doesFirstOptionHaveInput) {
```

Comment:
We need to still use `hideWhenJustOneOption` because it is used in getBookingResponsesSchema as well.

Also, this makes the code handle any other radioInput kind of custom field(when we allow that in future)